### PR TITLE
Split coremod and mod container to different classes

### PR DESCRIPTION
**Pull request by Prototik**
Don't be a jerk part 2.

Inspired by #2482 #2509 #2519 #2621 #2642 #2661. Tried to fix by #2604. Ignored in #2636. ~~All world against me.~~

Guys, seriously - it's fault of AE, not Sponge, Forge, FML, ~~Santa Claus~~ or someone else. IFMLLoadingPlugin should load minimal amount of classes to work properly. It's probably bad idea to load half of FML in coremod stage. Current sort of AppEngCore breaks separation of concerns rule at least, so if you don't respect me - respect your programming skill and don't ignore it, don't close it with no reason and don't shout "everyone do it wrong, that's work with clean forge so shut up". Be reasonable, please.

This PR don't break anything, but could save your time as you don't need to close 20 more tickets with a title "crash with sponge".

### CODE REVIEW

File: src/main/java/appeng/coremod/AppEngModContainer.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.coremod;
+
+
+import appeng.core.AEConfig;
+import com.google.common.eventbus.EventBus;
```
Comment (orod-org) [inline src/main/java/appeng/coremod/AppEngModContainer.java]: MINOR Wrong order for 'com.google.common.eventbus.EventBus' import. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.ImportOrderCheck)

File: src/main/java/appeng/coremod/AppEngModContainer.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import net.minecraftforge.fml.common.DummyModContainer;
```
Comment (orod-org) [inline src/main/java/appeng/coremod/AppEngModContainer.java]: MINOR 'net.minecraftforge.fml.common.DummyModContainer' should be separated from previous imports. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.ImportOrderCheck)

**Comment by orod-org** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by yueh** [comment]
As already said the goal is to remove the coremod completely, not fix it with a band-aid and forget it until it bites us again.

Should it not be possible, sure splitting it will be the option. But that should be the option of last resort, not the first one which allows us to forget about it again.

**Comment by Prototik** [comment]
Oh, that's problem? One momento.

**Comment by lucasdidur** [comment]
How long will it take for this problem to be fixed? Days, weeks, months, years?

**Comment by AnZaNaMa** [comment]
it's had only been 9 hours since he last said something. However, that doesn't mean that the team is going to accept his PR, which they have every right choose so. There is a big discussion on another issue somewhere here, but I don't have a link to it and it will tell you everything you need to know.

**Comment by humroben** [comment]
I was just looking this up in the issues, and I'm glad to see that this has already been acknowleged. I'm not worried about how long it takes, since letting mod developers work on it in their own time, allows them to make sure that it gets done right.

Keep up the good work

**Pull request closed without merging**